analyzers/tests/SonarAnalyzer.TestFramework/Verification/DiagnosticVerifier.cs

```diff
@@ -198,15 +203,26 @@ private static IEnumerable<IssueLocationPair> MatchDifferences(List<IssueLocatio
         {
             foreach (var actual in actualIssues.ToArray())  // First round removes all perfect matches, so we don't mismatch possible perfect match by imperfect one on the same line.
             {
-                if (expectedIssues.Remove(actual))
+                var expectedIndex = expectedIssues.IndexOf(actual);
+                if (expectedIndex >= 0)
                 {
+                    actual.UpdatePrimaryIssueIdFrom(expectedIssues[expectedIndex]);
+                    expectedIssues.RemoveAt(expectedIndex);
                     actualIssues.Remove(actual);
                 }
             }
             foreach (var actual in actualIssues)
             {
-                var expected = expectedIssues.OrderBy(x => Math.Abs(actual.Start ?? 0 - x.Start ?? 0)).ThenBy(x => Math.Abs(actual.Length ?? 0 - x.Length ?? 0)).FirstOrDefault();
-                expectedIssues.Remove(expected);
+                var expected = expectedIssues
+                    .OrderBy(x => actual.IssueId == x.IssueId ? 0 : 1)
+                    .ThenBy(x => Math.Abs(actual.Start ?? 0 - x.Start ?? 0))
+                    .ThenBy(x => Math.Abs(actual.Length ?? 0 - x.Length ?? 0))
+                    .FirstOrDefault();
+                if (expected is not null)
+                {
+                    actual.UpdatePrimaryIssueIdFrom(expected);
+                    expectedIssues.Remove(expected);
+                }
                 yield return new(actual, expected); // expected can be null
```

Comment:
Do you think we need this comment?
Right above you check nullability, and afterwards you unconditionally yield.
So it seems obvious that both null/not-null are valid.